services/command-router-base/src/main/java/org/eclipse/hono/commandrouter/impl/CommandRouterServiceImpl.java

```diff
@@ -207,55 +219,103 @@ private Duration getSanitizedLifespan(final Duration lifespan) {
 
     @Override
     public Future<CommandRouterResult> enableCommandRouting(final List<String> tenantIds, final Span span) {
+
+        if (!running.get()) {
+            return Future.succeededFuture(CommandRouterResult.from(HttpURLConnection.HTTP_UNAVAILABLE));
+        }
+
         Objects.requireNonNull(tenantIds);
         final boolean isProcessingRequired = tenantsToEnable.isEmpty();
-        tenantsToEnable.addAll(tenantIds);
+        tenantIds.stream()
+            .filter(s -> !reenabledTenants.contains(s))
+            .filter(s -> !tenantsInProcess.contains(s))
+            .filter(s -> tenantsToEnable.stream()
+                .allMatch(entry -> !entry.one().equals(s)))
+            .forEach(s -> tenantsToEnable.addLast(Pair.of(s, 1)));
+
         if (isProcessingRequired) {
             final Span processingSpan = tracer.buildSpan("re-enable command routing for tenants")
-                    .addReference(References.FOLLOWS_FROM, span.context())
+                    .addReference(References.CHILD_OF, span.context())
                     .start();
-            processTenantQueue(new ConcurrentHashSet<>(), processingSpan);
+            processingSpan.log(Map.of("number of tenant IDs", tenantIds.size()));
+            // make sure to report the span in a timely fashion
+            // so that it can be seen in the tracing back end
+            processingSpan.finish();
```

Comment:
I think this span isn't really needed - there is already the "enable command routing" parent span, with a `no_of_tenants` log entry.

Reply:
I have pushed corresponding changes ...